Tests/test_pile_ui.gd

```diff
@@ -1,49 +1,31 @@
-extends GutTest
+extends TestCardBase
 #test for ui pop up
 
-var _card_container_scene: PackedScene = load("res://Cards/CardContainer.tscn")
 var _card_container_scroll_scene: PackedScene  = load("res://#Scenes/CardScrollUI.tscn")
-var _card_container: CardContainer = null
 var _card_scroll: Control = null
 
-const SCROLL_CONTAINER_INDEX: int = 2
-const GRID_CONTAINER_INDEX: int = 0
-
-func before_all():
-	_card_container = _card_container_scene.instantiate()
-	# fill deck with 50 default cards
-	_card_container.default_deck.resize(50)
-	_card_container.max_hand_size = 10
-	_card_container.starting_hand_size = 0
-	_card_container.card_draw_time = .05
 
+# @Override
 func before_each():
+	super()
+	_card_scroll = _card_container_scroll_scene.instantiate()
+	_card_scroll.deck_pile = _card_container.default_deck.duplicate()
+	_card_scroll.populate("DeckPile")
 
-	# assign cards names of Card1-Card50
-	for card_index: int in _card_container.default_deck.size():
-		_card_container.default_deck[card_index] = CardBase.new()
-		_card_container.default_deck[card_index].card_title = "Card" + str(card_index + 1)
-	
-	get_tree().root.add_child(_card_container)
-	
 
+# @Override
 func after_each():
-	_card_container.free()
+	_card_container.queue_free()
```

Comment:
Since you inherit from TestCardBase, we should be able to call the super to free up the _card_container instead of calling it here, right?

We can also free the `_card_scroll` before we call the super, and remove that assert_no_new_orphans() to let the parent call that assert.

Reply:
the reason for that is that in `TestCardBase` you have:
```GDScript
func after_each():
	_card_container.queue_free()
	assert_no_new_orphans("Orphans still exist, please free up test resources.")
```
and In `test_pile_ui`:
```GDScript
func after_each():
	_card_container.queue_free()
	_card_scroll.free()
	assert_no_new_orphans("Orphans still exist, please free up test resources.")
```
If I was to call `super` in `test_pile_ui`, the `assert_no_new_orphans` would be called before I have a chance to call the `_card_scroll.free()` which is relevant only to this test. Which is why I need to rewrite the `after_each` instead of calling `super`

Reply:
Ah I get ya now. We'd have to recall it again. Sounds good!